### PR TITLE
fix android pack build

### PR DESCRIPTION
**Pull request by alalek**

### CODE REVIEW

File: modules/core/src/convert.cpp [hunk 1]
```diff
@@ -4585,7 +4585,7 @@ cvtScaleHalf_<float, short>( const float* src, size_t sstep, short* dst, size_t
                     __m128i v_dst = _mm_cvtps_ph(v_src, 0);
 
                     _mm_storel_epi64((__m128i *)(dst + x), v_dst);
-#elif defined __GNUC__ && (defined __arm__ || defined __aarch64__)
+#elif defined __GNUC__ && (defined CV_NEON || defined __aarch64__)
```
Comment (ilya-lavrenov) [inline modules/core/src/convert.cpp]: `defined CV_NEON` -> `CV_NEON`?

Comment (alalek) [inline modules/core/src/convert.cpp]: @ilya-lavrenov Right. But there is runtime (logic) error also, it is just a test.
Could you add some links with information about compiler requirements for float16x4_t ?
```
/opt/android/android-ndk-r9/toolchains/arm-linux-androideabi-4.8/prebuilt/linux-x86_64/bin/arm-linux-androideabi-g++ -mfpu=neon -mfp16-format=ieee -mfpu=neon-fp16 -dM -E - <<< "" | grep ARM
#define __ARM_SIZEOF_WCHAR_T 32
#define __ARM_ARCH_ISA_ARM 1
#define __ARMEL__ 1
#define __ARM_ARCH_5TE__ 1
#define __ARM_FP 14
#define __ARM_NEON_FP 6
#define __ARM_SIZEOF_MINIMAL_ENUM 4
#define __ARM_PCS 1
#define __ARM_FP16_FORMAT_IEEE 1
#define __ARM_FEATURE_QBIT 1
#define __ARM_FEATURE_CLZ 1
#define __ARM_ARCH_ISA_THUMB 1
#define __ARM_ARCH 5
#define __ARM_EABI__ 1
#define __ARM_FEATURE_DSP 1
```
Error log is [here](http://pullrequest.opencv.org/buildbot/builders/precommit_pack_android/builds/107/steps/build%20sdk/logs/stdio)

Comment (tomoaki0705) [inline modules/core/src/convert.cpp]: Curious....
float16x4_t is usually defined in arm_neon.h.

```
odroid@odroid:~$ ls /usr/lib/gcc/arm-linux-gnueabihf/4.8.4/include/arm_neon.h
/usr/lib/gcc/arm-linux-gnueabihf/4.8.4/include/arm_neon.h
odroid@odroid:~$ grep float16x4_t !$
grep float16x4_t /usr/lib/gcc/arm-linux-gnueabihf/4.8.4/include/arm_neon.h
typedef __builtin_neon_hf float16x4_t   __attribute__ ((__vector_size__ (8)));
```

It depends on the compiler, so I was suspecting that this definition was guarded by ```#if ((__ARM_FP & 0x2) != 0``` and __ARM_FP has 12 (i.e. mfpu=neon is accepted, but not the mfpu=neon-fp16)
In @alalek log, it states that ```__ARM_FP``` is 14 and ```__ARM_NEON_FP``` is 6, which means, mfpu=neon-fp16 is clearly passed.
I wonder why.

Comment (alalek) [inline modules/core/src/convert.cpp]: In Android NDK r9 llvm toolchans has float16x4_t and vcvt_f32_f16 in headers, but mentioned bits in flags are fine.
```
$ grep -R vcvt_f32_f16 /opt/android/android-ndk-r9/
/opt/android/android-ndk-r9/toolchains/llvm-3.2/prebuilt/linux-x86_64/lib/clang/3.2/include/arm_neon.h:__ai float32x4_t vcvt_f32_f16(float16x4_t __a) {
/opt/android/android-ndk-r9/toolchains/llvm-3.2/prebuilt/linux-x86_64/lib/clang/3.2/include/arm_neon.h:  return (float32x4_t)__builtin_neon_vcvt_f32_f16((int8x8_t)__a, 6); }
Binary file /opt/android/android-ndk-r9/toolchains/llvm-3.3/prebuilt/linux-x86_64/bin/clang++ matches
Binary file /opt/android/android-ndk-r9/toolchains/llvm-3.3/prebuilt/linux-x86_64/bin/clang matches
/opt/android/android-ndk-r9/toolchains/llvm-3.3/prebuilt/linux-x86_64/lib/clang/3.3/include/arm_neon.h:__ai float32x4_t vcvt_f32_f16(float16x4_t __a) {
/opt/android/android-ndk-r9/toolchains/llvm-3.3/prebuilt/linux-x86_64/lib/clang/3.3/include/arm_neon.h:  return (float32x4_t)__builtin_neon_vcvt_f32_f16((int8x8_t)__a, 6); }
```

I will try to use CMake "try_compile" here.

**Comment by alalek** [comment]
:+1:

**Pull request merged**